### PR TITLE
metrics: support multi-k8s in grafana dashboards

### LINKED ISSUE #4673: support multi-k8s in grafana dashboards
## Enhancement Task

K8s community and organizations are increasingly deploying multiple Kubernetes clusters to improve availability, isolation and scalability. Since TiDB Operator have support [deploying tidb across-kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/dev/deploy-tidb-cluster-across-multiple-kubernetes), we propose to add `k8s_cluster` label to address the multi-k8s issue without affecting existing single-cluster use case.

### PR DESCRIPTION
**Pull request by just1900**
Signed-off-by: just1900 <legendj228@gmail.com>

### What problem does this PR solve?

Issue Number: close #4673

K8s community and organizations are increasingly deploying multiple Kubernetes clusters to improve availability, isolation and scalability. Since TiDB Operator have support [deploying tidb across-kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/dev/deploy-tidb-cluster-across-multiple-kubernetes), this PR will address the multi-k8s issue without affecting existing single-cluster use case.

### What is changed and how it works?

What's Changed:

* add a tidb_cluster label in all expr
* add cluster variable in Grafana templating

How it works:

* For users with single-k8s, nothing need to be changed, just use it as before.
* For users with multi-k8s:
  * first add `kubernetes` label to identify k8s cluster in your prometheus' `external_labels`.
  * set field "hide" to '0' to show variables in grafana via `sed -i s/"hide": 2/"hide": 0/g <path/to/dashboard.json>`
  * load it to grafana

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes
- tidb https://github.com/pingcap/tidb/pull/32546
- tikv https://github.com/tikv/tikv/pull/12008
- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
metrics: support multi-k8s in grafana dashboards
```

### CODE REVIEW

**Comment by nolouch** [comment]
Could you add an issue link? and there must reference other components.

**Comment by rleungx** [comment]
Do we have any test of this PR?

**Comment by just1900** [comment]
> Could you add an issue link? and there must reference other components.

I've added PRs for other components into description.

> Do we have any test of this PR?

Manual testet by load it to my grafana, it works fine.

**Review by nolouch — APPROVED** [review]

File: metrics/grafana/pd.json [hunk 1]
```diff
@@ -134,7 +134,7 @@
       "tableColumn": "",
       "targets": [
         {
-          "expr": "pd_tso_role{tidb_cluster=\"$tidb_cluster\", instance=\"$instance\", dc=\"global\"}",
+          "expr": "pd_tso_role{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=\"$instance\", dc=\"global\"}",
```
Comment (rleungx) [inline metrics/grafana/pd.json:137]: Will it cause tiup cannot display?

Comment (just1900) [reply metrics/grafana/pd.json:137]: If `k8s_cluster` is empty, then the condition will be like `k8s_cluster=""`, which is the same as a no op in prometheus.

**Review by rleungx — APPROVED** [review]

**Pull request merged**